compiler/vm/src/heap/object_representation.md

```diff
@@ -0,0 +1,137 @@
+# Heap Object Representation
+
+The lowest granularity for objects in Candy's memory (stack and heap) is one _word_: 64 bit = 8 byte.
+An object is stored as an _inline object_:
+
+## Inline Object
+
+An inline object is a single word containing a tagged union of different types of values:
+
+|                                                                     Value | Meaning     |
+| ------------------------------------------------------------------------: | :---------- |
+| `xxxxxxxx xxxxxxxx xxxxxxxx xxxxxxxx xxxxxxxx xxxxxxxx xxxxxxxx xxxxx000` | Pointer     |
+| `xxxxxxxx xxxxxxxx xxxxxxxx xxxxxxxx xxxxxxxx xxxxxxxx xxxxxxxx xxxxxx01` | Int         |
+| `xxxxxxxx xxxxxxxx xxxxxxxx xxxxxxxx xxxxxxxx xxxxxxxx xxxxxxxx xxxxx010` | SendPort    |
+| `xxxxxxxx xxxxxxxx xxxxxxxx xxxxxxxx xxxxxxxx xxxxxxxx xxxxxxxx xxxxx110` | ReceivePort |
+| `xxxxxxxx xxxxxxxx xxxxxxxx xxxxxxxx xxxxxxxx xxxxxxxx xxxxxxxx xxxxxx11` | Builtin     |
+
+> The remaining patterns are reserved for future use.
```

Comment:
There aren't really any remaining patterns, the ones above are exhaustive.
And I believe we don't make any guarantees about stability of the format anyways, so I'd just remove this.